### PR TITLE
Feature/secrets

### PR DESCRIPTION
**Pull request by andrewstevenson**
This change is [Reviewable](https://reviewable.io/reviews/landoop/lenses-go/6)

### CODE REVIEW

File: cmd/lenses-cli/configuration.go [hunk 1]
```diff
@@ -99,6 +99,14 @@ func newConfigurationManager(set *pflag.FlagSet) *configurationManager {
 	return m
 }
 
+func newEmptyConfigManager() *configurationManager {
```
Comment (spirosoik) [inline cmd/lenses-cli/configuration.go]: I see it is used only in one place, so may a method for this is redundant. As I understand your intent was to make it re-usable right?

File: cmd/lenses-cli/main.go [hunk 2]
```diff
@@ -38,12 +38,12 @@ const (
...
-		Name:            "lenses-cli",
-		Description:     "Lenses-cli is the command line client for the Lenses REST API.",
-		Version:         lenses.Version,
-		PersistentFlags: setupConfigManager,
-		ShowSpinner:     false,
-		Setup:           setup,
+		Name:        "lenses-cli",
+		Description: "Lenses-cli is the command line client for the Lenses REST API.",
+		Version:     lenses.Version,
+		//PersistentFlags: setupConfigManager,
```
Comment (spirosoik) [inline cmd/lenses-cli/main.go]: you have commented it but I see you use it here https://github.com/Landoop/lenses-go/pull/6/files#diff-514e19b9501e58b1a5f1a32e1be48862R136

File: secrets.go [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+			golog.Infof("Found environment var [%s]", v)
+			secretVars = append(secretVars, strings.Replace(v, "SECRET_", "", -1))
+		}
+	}
+
+	return secretVars
+}
+
+func getVaultClient(server, token string) (*vaultapi.Client, error) {
+	client, err := vaultapi.NewClient(&vaultapi.Config{
```
Comment (spirosoik) [inline secrets.go]: why not set the token here and you are setting it after the error check?

File: secrets.go [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+	return secrets, nil
+}
+
+// NewKeyVaultClient creates a new keyvault client
+func newKeyVaultClient(vaultURL string, config AzureConfiguration) (*KeyVault, error) {
+
+	keyClient := keyvault.New()
+	credentials := azureAuth.NewClientCredentialsConfig(config.ClientID, config.ClientSecret, config.TenantID)
+	// set the correct resource for keyvault
+	credentials.Resource = "https://vault.azure.net"
```
Comment (spirosoik) [inline secrets.go]: this could be a `const` instead or may even to be configurable via an environment or file. May later will change the resource

**Comment by spirosoik** [comment]
@andrewstevenson also before merge let's keep one commit which is related with the feature we add.

File: cmd/lenses-cli/configuration.go (hunk 1 above)
Comment (andrewstevenson) [reply cmd/lenses-cli/configuration.go]: Removing it.

File: secrets.go (hunk 4 above)
Comment (andrewstevenson) [reply secrets.go]: Done

File: secrets.go (hunk 3 above)
Comment (andrewstevenson) [reply secrets.go]: I don't understand?

File: cmd/lenses-cli/main.go (hunk 2 above)
Comment (andrewstevenson) [reply cmd/lenses-cli/main.go]: I set this later in main.go as we don't need the default flags for secrets as we don't speak to Lenses so no need for kerberos etc.

File: secrets.go (hunk 3 above)
Comment (spirosoik) [reply secrets.go]: here you set the token https://github.com/Landoop/lenses-go/pull/6/files#diff-f312e33df50115b7e158e79a3e55d96bR89


but if you have `VAULT_TOKEN` in environment file this is set automatically. https://github.com/hashicorp/vault/blob/master/api/client.go#L342

**Pull request closed without merging**